docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,330 @@
+
+### Running Tests Locally
+
+Run these commands from your component/pipeline directory before submitting your contribution:
+
+```bash
+# Run all unit tests with coverage reporting
+pytest --cov=src --cov-report=html
+
+# Run specific test files when debugging
+pytest tests/test_my_component.py -v
+```
+
+### Code Quality Checks
+
+Ensure your code meets quality standards:
+
+```bash
+# Format checking (120 character line length)
+black --check --line-length 120 .
+
+# Docstring validation (Google convention)
+pydocstyle --convention=google .
+
+# Validate metadata schema
+python scripts/validate_metadata.py
+
+# Run all pre-commit hooks
+pre-commit run --all-files
+```
+
+### Building Custom Container Images
+
+If your component uses a custom image, test the container build:
+
+```bash
+# Build your component image
+docker build -t my-component:test components/<category>/my-component/
+
+# Test the container runs correctly
+docker run --rm my-component:test --help
+```
+
+### CI Pipeline
+
+GitHub Actions automatically runs these checks on every pull request:
+
+- Code formatting (Black), linting (Flake8), docstring validation (pydocstyle), type checking (MyPy)
+- Unit and integration tests with coverage reporting
+- Container image builds for components with Containerfiles
+- Security vulnerability scans
+- Metadata schema validation
+- Standardized README content and formatting conformance
+
+## Submitting Your Contribution
+
+### Commit Your Changes
+
+Use descriptive commit messages following the [Conventional Commits](https://conventionalcommits.org/) format:
+
+```bash
+git add .
+git status  # Review what you're committing
+git diff --cached  # Check the actual changes
+
+git commit -m "feat(training): add <my_component> training component
+
+- Implements <my_component> Core-Tier component
+- Includes comprehensive unit tests with 95% coverage
+- Provides working pipeline examples
+- Resolves #123"
+```
+
+### Push and Create Pull Request
+
+Push your changes and create a pull request on GitHub:
+
+```bash
+git push origin component/my-component
+```
+
+On GitHub, click "Compare & pull request" and fill out the PR template provided with appropriate details
+
+All PRs must pass:
+- Automated checks (linting, tests, builds)
+- Code review by maintainers and community members
+- Documentation review
+
+### Pre-Submission Checklist
+
+Verify your contribution meets all requirements before requesting review:
+
+- [ ] All tests pass
+- [ ] Pre-commit hooks pass without errors
+- [ ] All required files are present and complete
+- [ ] `metadata.yaml` includes fresh `lastVerified` timestamp
+- [ ] OWNERS file lists appropriate maintainers
+- [ ] README provides clear documentation with usage examples
+- [ ] Component follows `snake_case` naming convention
+- [ ] (Core tier only) Containerfile present if using custom image
+- [ ] No security vulnerabilities in dependencies
+- [ ] Container builds successfully (if applicable)
```

Comment:
this should be a checklist in the PR template it self and not added here, otherwise we have to update two places everytime the pr template is updated

Reply:
okay, just ensure it's captured in some Jira or somewhere so we don't forget